src/NuGet.Core/NuGet.Versioning/VersionFormatter.cs

```diff
@@ -27,49 +27,25 @@ public string Format(string format, object arg, IFormatProvider formatProvider)
                 throw new ArgumentNullException(nameof(arg));
             }
 
-            string formatted = null;
-            var argType = arg.GetType();
-
-            if (argType == typeof(IFormattable))
+            if (string.IsNullOrEmpty(format) || arg is not SemanticVersion version)
             {
-                formatted = ((IFormattable)arg).ToString(format, formatProvider);
+                return null;
             }
-            else if (!String.IsNullOrEmpty(format))
+
+            var builder = StringBuilderPool.Shared.Rent(Math.Max(format.Length, 100));
```

Comment:
I picked the largest that I would think would exist given all the different stuff you can shove in there. The number is actually irreverent assuming its below 256 because the pool will always return the same size. I can change this number to whatever you'd like. :)

Reply:
Ah, totally forgot about that part. 
Looks good other than that.

We can do 2 things:

* Isolate the changes in this PR, removing the bytes array stuff.
* Wait for the byte array PR to complete before rebasing. 

There's some formatting failures reported (no test issues). 

```
  src\NuGet.Core\NuGet.Versioning\VersionFormatter.cs(47,1): Fix whitespace formatting.
  Complete in 7248ms.
  Formatted code file 'C:\a\1\s\src\NuGet.Core\NuGet.Versioning\VersionFormatter.cs'.
```

Reply:
I'll wait for the byte array change, and then rebase.

Reply:
@nkolev92 Can you turn on whitespace analyzers so that I found this out while I'm coding, rather than after I've submitted? :)

Reply:
That was harder to figure out than it should've been. 
Not sure *why* the build is not failing automatically, but the IDE surfaces the issues: 

https://github.com/NuGet/NuGet.Client/pull/4138

Reply:
If there is no significance for `100` then why not `StringBuilderPool.Shared.Rent(format.Length); ` this? The reason is `Rent()` returns a `StringBuilder` instance that can accommodate maximum of 256 chars.

Reply:
Why do you think that `format.Length` would make more sense?

``` C#
formatter.Format("N", version);
```

Format.Length has a length of 1, but produces a string in the format of "1.2.3-pre+c76e3ff23fd58bf820457aecc76880f79aec6ac4".

Reply:
> Format.Length has a length of 1, but produces a string in the format of "1.2.3-pre+c76e3ff23fd58bf820457aecc76880f79aec6ac4".

I was not aware of this context. thanks for clarifying.